src/visitor.rs

```diff
@@ -703,8 +704,22 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
             .collect();
 
         if items.is_empty() {
-            self.visit_stmt(&stmts[0]);
-            self.walk_stmts(&stmts[1..]);
+            // The `if` expression at the end of the block should be formatted in a single
+            // line if possible.
+            if self.config.version() == Version::Two
+                && stmts.len() == 1
+                && crate::expr::stmt_is_if(&stmts[0])
+                && !contains_skip(get_attrs_from_stmt(&stmts[0]))
+            {
+                let shape = self.shape();
+                let rewrite = self.with_context(|ctx| {
+                    format_expr(stmt_expr(&stmts[0])?, ExprType::SubExpression, ctx, shape)
```

Comment:
This is in theory applicable to any expression kind rather than to `if` only.
Basically, the rule is "print trailing expressions as expressions, rather than statements".

Reply:
I'm not sure where else that would make a difference though.

Reply:
Thanks, I think that's right. The only thing to take into account is if the last expression is one of `continue`, `return` or `break`, then we need to add a trailing semicolon. I'll see if I can simplify this a bit more.